arch/arm/src/common/Toolchain.defs

```diff
@@ -227,6 +227,13 @@ else
       ARCHOPTIMIZATION += -fno-builtin
     endif
   endif
+
+  # Skip -Warray-bounds check due to bug of GCC-12.2:
+  # Wrong warning array subscript [0] is outside array bounds:
+  # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105523
+
```

Comment:
Instead of disabling this globally since it is a good check, can we do as the ticket suggests and set

```
--param=min-pagesize=0 
```

As needed?

Reply:
Ok, min-pagesize is a newly added option of GCC-12. It seems that most open source projects use this option for workaround. I think we can add gcc version checks to avoid breaks on other versions of the compiler.

Reply:
Done